scripts/temp_pub.py

```diff
@@ -12,17 +12,28 @@ def internal_temperature_publisher():
     msg_temp = Temperature()
     msg_temp.header.frame_id = "internal_temperature_sensor_link"
     rate = rospy.Rate(10)
+    prev_temp = 0
+
     while not rospy.is_shutdown():
         try:
             msg_temp.header.stamp = rospy.Time.now()
             sensor_data = temper.get_temperature()
-            msg_temp.temperature = float(sensor_data[0])
+
+            if sensor_data is not None and sensor_data[0] is not None:
+                msg_temp.temperature = float(sensor_data[0])
+                prev_temp = float(sensor_data[0])
+            else:
+                msg_temp.temperature = float(prev_temp)
+                rospy.loginfo("Error reading temperature, displaying previously read temperature")
+            
+            print(msg_temp)
```

Comment:
remove print state. I would recommend using rospy.logdebug for debugging purpose